server/src/main/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImpl.java

```diff
@@ -942,7 +942,20 @@ private UserVm importVirtualMachineInternal(final UnmanagedInstanceTO unmanagedI
         if (CollectionUtils.isEmpty(unmanagedInstanceDisks)) {
             throw new ServerApiException(ApiErrorCode.INTERNAL_ERROR, String.format("No attached disks found for the unmanaged VM: %s", instanceName));
         }
-        final UnmanagedInstanceTO.Disk rootDisk = unmanagedInstance.getDisks().get(0);
+        UnmanagedInstanceTO.Disk rootDisk = unmanagedInstance.getDisks().get(0);
```

Comment:
@DK101010 Since disks list for UnmanagedInstance is created by CloudStack only, will it be better to always prepare that list in a way that first disk is always ROOT disk.
Relevant code can be found in VmwareResource class.

Reply:
@shwstppr ahh ok, yes of course this is a better solution, i will check this.